consensus/istanbul/backend/api.go

```diff
@@ -214,41 +214,61 @@ func (api *APIExtension) GetCommitteeSize(number *rpc.BlockNumber) (int, error)
 	}
 }
 
-func (api *APIExtension) getProposerAndValidators(block *types.Block) (common.Address, []common.Address, error) {
+type ConsensusInfo struct {
+	proposer       common.Address
+	originProposal common.Address
+	committee      []common.Address
+	round          byte
+}
+
+func (api *APIExtension) getConsensusInfo(block *types.Block) (ConsensusInfo, error) {
 	blockNumber := block.NumberU64()
 	if blockNumber == 0 {
-		return common.Address{}, []common.Address{}, nil
+		return ConsensusInfo{}, nil
 	}
 
+	round := block.Header().Round()
 	view := &istanbul.View{
 		Sequence: new(big.Int).Set(block.Number()),
-		Round:    new(big.Int).SetInt64(0),
+		Round:    new(big.Int).SetInt64(int64(round)),
 	}
 
 	// get the proposer of this block.
 	proposer, err := ecrecover(block.Header())
 	if err != nil {
-		return common.Address{}, []common.Address{}, err
+		return ConsensusInfo{}, err
 	}
 
 	// get the snapshot of the previous block.
 	parentHash := block.ParentHash()
 	snap, err := api.istanbul.snapshot(api.chain, blockNumber-1, parentHash, nil)
 	if err != nil {
-		return proposer, []common.Address{}, err
+		return ConsensusInfo{}, err
+	}
+
+	// get origin proposer at 0 round.
+	originProposer := common.Address{}
+	lastProposer := common.Address{}
+	// TODO-Klaytn add the logic to get the last proposer for other policy. Weighted round robin doesn't need it.
+	if istanbul.ProposerPolicy(snap.Policy) == istanbul.WeightedRandom {
+		newValSet := snap.ValSet.Copy()
+		newValSet.CalcProposer(lastProposer, 0)
+		originProposer = newValSet.GetProposer().Address()
```

Comment:
Is there any reason for coping valSet??
What about just calling CalcProposer and GetProposer functions without coping?

Reply:
I worried that `CalcProposer` can change the result of `SubListWithProposer`.
If it is not, I will remove the copy logic.
Is it ok? I will look`SubListWithProposer` more. :)

Reply:
For now, It looks like that `CalcProposer` doesn't make any effect to `SubListWithProposer`. But how about separate `snap.valset` for round 0 and other.

Reply:
Yes CalcProposer has no effect to SubListWithProposer function for now.
But It would be better separating them, just for sure.